### PR TITLE
bokeh/bokeh#11640 Document Tool.icon in user guide

### PR DESCRIPTION
**Pull request by ABODFTW**
Added documentation based on #11640, had a problem with adding icons from an URL, tried PNG icons it gave the following error `icon given the invalid value` in the dev tool.

Tried passing a PIL Image object, and it gave me this error `Object of type Image is not JSON serializable`

So I need help figuring out if the way I'm trying to change the icon is correct, and if you got any feedback on the PR.

### CODE REVIEW

**Comment by bryevdv** [comment]
@ABODFTW that feature only exists on `branch-3.0` at present, it's not yet in any release. Are you testing with a build from source?

**Comment by ABODFTW** [comment]
Yes I'm building from source on 3.0, here is what I get when running `bokeh info`

```
Python version      :  3.9.6 | packaged by conda-forge | (default, Jul  6 2021, 08:53:59)
IPython version     :  7.29.0
Tornado version     :  6.1
Bokeh version       :  3.0.0dev1+21.g51a06bb37.dirty
BokehJS static path :  /home/laptop/projects/bokeh/bokeh/server/static
node.js version     :  v16.6.1
npm version         :  7.20.3
```

**Comment by bryevdv** [comment]
Might be a regression or bug then, cc @mattpap PIL is supposed to work now?

**Comment by mattpap** [comment]
It doesn't work, because transformations are applied only to top-level types, but not in case of composite types, like `Either`, which is used in `Tool.icon`, so `bokeh.core.property.Image.transfom` is never called.

**Comment by mattpap** [comment]
@ABODFTW, please start an issue for the broken functionality.

**Comment by ABODFTW** [comment]
Done, created #11830

**Comment by bryevdv** [comment]
Thanks for the issue @ABODFTW For this PR would you prefer to:

* keep it open as-is until #11830 is fixed and merge later, or
* Update to remove PIL references, and merge now

It's really whatever you would prefer

File: sphinx/source/docs/user_guide/tools.rst [hunk 1]
```diff
@@ -111,6 +111,49 @@ method with the figure() function's tools argument:
...
+
+.. code-block:: python
+
+    plot.add_tools(BoxSelectTool(description="My tool"))
+
+
+It's also possible to change a tool icon using the ``icon`` keyword.
+
+You can pass:
+    1. A well known icon name
```
Comment (bryevdv) [inline sphinx/source/docs/user_guide/tools.rst:131]:
```suggestion
You can pass:

    1. A well known icon name
```

File: sphinx/source/docs/user_guide/tools.rst [hunk 2]
```diff
@@ -111,6 +111,49 @@ method with the figure() function's tools argument:
...
+
+        .. code-block:: python
+
+            plot.add_tools(BoxSelectTool(icon="box_zoom"))
+    2. A CSS selector
```
Comment (bryevdv) [inline sphinx/source/docs/user_guide/tools.rst:136]:
```suggestion
            plot.add_tools(BoxSelectTool(icon="box_zoom"))
            
    2. A CSS selector
```

File: sphinx/source/docs/user_guide/tools.rst [hunk 3]
```diff
@@ -111,6 +111,49 @@ method with the figure() function's tools argument:
...
+
+        .. code-block:: python
+
+            plot.add_tools(BoxSelectTool(icon=".my-icon-class"))
+    3. An image path
```
Comment (bryevdv) [inline sphinx/source/docs/user_guide/tools.rst:141]:
```suggestion
            plot.add_tools(BoxSelectTool(icon=".my-icon-class"))
            
    3. An image path
```

File: sphinx/source/docs/user_guide/tools.rst [hunk 4]
```diff
@@ -111,6 +111,49 @@ method with the figure() function's tools argument:
...
+
+        .. code-block:: python
+
+            plot.add_tools(BoxSelectTool(icon="path/to/icon"))
+    4. A PIL Image instance
```
Comment (bryevdv) [inline sphinx/source/docs/user_guide/tools.rst:146]:
```suggestion
            plot.add_tools(BoxSelectTool(icon="path/to/icon"))
            
    4. A PIL Image instance
```

File: sphinx/source/docs/user_guide/tools.rst [hunk 5]
```diff
@@ -111,6 +111,49 @@ method with the figure() function's tools argument:
...
+----------------------
+You can change the :ref:`bokeh.models.tools` tooltip by passing it to
+the ``description`` keyword using the ``add_tools()`` method of a plot or any
+of it's instances like |figure|.
+
+.. code-block:: python
+
+    plot.add_tools(BoxSelectTool(description="My tool"))
+
+
```
Comment (bryevdv) [inline sphinx/source/docs/user_guide/tools.rst:127]:
```suggestion

```

**Review by bryevdv — COMMENTED** [review]
blank link after codeblocks is more consistent with usage throughout the rest of the docs

**Comment by bryevdv** [comment]
@ABODFTW I also left some very minor suggestions to make vertical whitespace more consistent with the rest of the docs ReST files

**Comment by ABODFTW** [comment]
@bryevdv Thanks for the suggestions, committed suggested edits and removed PIL reference.

File: sphinx/source/docs/user_guide/tools.rst [hunk 6]
```diff
@@ -111,6 +111,40 @@ method with the figure() function's tools argument:
...
+Customizing tools icon
+----------------------
+You can change the :ref:`bokeh.models.tools` tooltip by passing it to
+the ``description`` keyword using the ``add_tools()`` method of a plot or any
+of it's instances like |figure|.
+
+.. code-block:: python
+
+    plot.add_tools(BoxSelectTool(description="My tool"))
+It's also possible to change a tool icon using the ``icon`` keyword.
```
Comment (bryevdv) [inline sphinx/source/docs/user_guide/tools.rst:125]:
```suggestion
    plot.add_tools(BoxSelectTool(description="My tool"))

It's also possible to change a tool icon using the ``icon`` keyword.
```

Sphinx demands a blank line after indented blocks

File: sphinx/source/docs/user_guide/tools.rst [hunk 7]
```diff
@@ -111,6 +111,40 @@ method with the figure() function's tools argument:
...
+
+You can pass:
+
+    1. A well known icon name
+
+        .. code-block:: python
+
+            plot.add_tools(BoxSelectTool(icon="box_zoom"))
+
```
Comment (bryevdv) [inline sphinx/source/docs/user_guide/tools.rst:134]:
```suggestion

```
trailing whitespace

File: sphinx/source/docs/user_guide/tools.rst [hunk 8]
```diff
@@ -111,6 +111,40 @@ method with the figure() function's tools argument:
...
+        .. code-block:: python
+
+            plot.add_tools(BoxSelectTool(icon="box_zoom"))
+            
+    2. A CSS selector
+
+        .. code-block:: python
+
+            plot.add_tools(BoxSelectTool(icon=".my-icon-class"))
+
```
Comment (bryevdv) [inline sphinx/source/docs/user_guide/tools.rst:140]:
```suggestion

```
trailing whitespace

**Comment by bryevdv** [comment]
Thanks @ABODFTW there are some whitespace issues that trip up both the linter and Sphinx, I've added suggestions that should resolve them.

**Comment by ABODFTW** [comment]
Thanks a lot for being patient with me, really appreciate it, committed new suggestions, is it mergeable now?

File: sphinx/source/docs/user_guide/tools.rst [hunk 9]
```diff
@@ -111,6 +111,39 @@ method with the figure() function's tools argument:
...
+It's also possible to change a tool icon using the ``icon`` keyword.
+
+You can pass:
+
+    1. A well known icon name
+
+        .. code-block:: python
+
+            plot.add_tools(BoxSelectTool(icon="box_zoom"))
+    2. A CSS selector
```
Comment (bryevdv) [inline sphinx/source/docs/user_guide/tools.rst:135]:
```suggestion
            plot.add_tools(BoxSelectTool(icon="box_zoom"))

    2. A CSS selector
```

File: sphinx/source/docs/user_guide/tools.rst [hunk 10]
```diff
@@ -111,6 +111,39 @@ method with the figure() function's tools argument:
...
+
+        .. code-block:: python
+
+            plot.add_tools(BoxSelectTool(icon=".my-icon-class"))
+    3. An image path
```
Comment (bryevdv) [inline sphinx/source/docs/user_guide/tools.rst:140]:
```suggestion
            plot.add_tools(BoxSelectTool(icon=".my-icon-class"))

    3. An image path
```

**Comment by bryevdv** [comment]
@ABODFTW almost! My suggestions were to remove trailing whitespace on some "blank" lines, but you removed the entire lines, which Sphinx does not like because code blocks must have an  (empty, no trailing whitespace) line after the block. If you want you can apply my suggestions above from the GH web UI

**Comment by ABODFTW** [comment]
Hopefully, that's how it should look like?

**Comment by bryevdv** [comment]
Thanks @ABODFTW the one test failure was an unrelated conda network issue, merging now!

**Pull request merged**